lib/maths/CBoostedTreeImpl.cc

```diff
@@ -414,6 +368,21 @@ void CBoostedTreeImpl::computeProbabilityAtWhichToAssignClassOne(const core::CDa
     }
 }
 
+void CBoostedTreeImpl::initializeTreeShap(const core::CDataFrame& frame) {
+    // Populate number samples reaching each node.
+    CTreeShapFeatureImportance::computeNumberSamples(m_NumberThreads, frame,
+                                                     *m_Encoder, m_BestForest);
+
+    if (m_NumberTopShapValues > 0) {
+        // Create the SHAP calculator.
+        m_TreeShap = std::make_unique<CTreeShapFeatureImportance>(
+            frame, *m_Encoder, m_BestForest, m_NumberTopShapValues);
+    } else {
+        // Set internal node values anyway.
+        CTreeShapFeatureImportance::computeInternalNodeValues(m_BestForest);
```

Comment:
At the moment, we don't have internal node values as a part of the inference model definition. IMO, we should either communicated them to java as well or omit to compute them if SHAP is not required.

Reply:
Good point! I think it would be quite nice to have them (independent of having feature importance) since they could be used for the basis of interesting tree visualisations. However, there is no point in having them if they aren't exposed. I'll comment this out and leave a comment to revisit.

Reply:
See [24d7c4e](https://github.com/elastic/ml-cpp/pull/1023/commits/24d7c4e3aa866f0c5edc25d971b8319a0cb4e2d4).